File: s-personas/src/main/java/co/edu/javeriana/arqui/rest/repository/RepositorioPacientes.java
package co.edu.javeriana.arqui.rest.repository;

import co.edu.javeriana.arqui.rest.entity.Paciente;
import jakarta.ejb.Stateless;
import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import jakarta.persistence.TypedQuery;
import java.util.List;

@Stateless
public class RepositorioPacientes {

    @PersistenceContext(unitName = "PersonasPU")
    private EntityManager em;

    public void crear(Paciente paciente) {
        em.persist(paciente);
    }

    public List<Paciente> listarTodos() {
        TypedQuery<Paciente> query = em.createQuery("SELECT p FROM Paciente p", Paciente.class);
        return query.getResultList();
    }
}

File: s-personas/src/main/java/co/edu/javeriana/arqui/rest/repository/RepositorioPacientes.kt
package co.edu.javeriana.arqui.rest.repository

import co.edu.javeriana.arqui.rest.entity.Paciente
import jakarta.ejb.Stateless
import jakarta.persistence.EntityManager
import jakarta.persistence.PersistenceContext

@Stateless
open class RepositorioPacientes {

    @PersistenceContext(unitName = "PersonasPU")
    private lateinit var em: EntityManager

    fun crear(paciente: Paciente) {
        em.persist(paciente)
    }

    fun listarTodos(): List<Paciente> {
        val query = em.createQuery("SELECT p FROM Paciente p", Paciente::class.java)
        return query.resultList
    }
}